### PR TITLE
[HOPSWORKS-643] Refactor TfServing code to allow pluggable providers

### PR DESCRIPTION
**Pull request by SirOibaf**
## Make sure there is no duplicate PR for this issue

* **Please check if the PR meets the following requirements**
- [x] Adds tests for the submitted changes (for bug fixes & features)
- [x] Passes the tests
- [x] HOPSWORKS JIRA issue has been opened for this PR
- [x] All commits have been squashed down to a single commit

* **Post a link to the associated JIRA issue**

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* **What is the new behavior (if this is a feature change)?**

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

* **Other information**:

### CODE REVIEW

File: hopsworks-api/src/main/java/io/hops/hopsworks/api/tensorflow/ServingConfView.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+package io.hops.hopsworks.api.tensorflow;
+
+import io.swagger.annotations.ApiModel;
+import io.swagger.annotations.ApiModelProperty;
+
+import javax.xml.bind.annotation.XmlRootElement;
+
+@XmlRootElement
+@ApiModel(value = "Represent configuration for serving UI")
+public class ServingConfView {
```
[the remark this answers is not in the document]
Comment (SirOibaf) [reply hopsworks-api/src/main/java/io/hops/hopsworks/api/tensorflow/ServingConfView.java]: At this stage yes, the view contains just an integer. However in the future, we'll expand this view to contain more info (e.g. Is sklearn serving enabled?) stuff like that. 
I'm not sure what's the best way of implementing this. I'm open to suggestions

File: hopsworks-api/src/main/java/io/hops/hopsworks/api/tensorflow/TfServingLifecycleView.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ *
+ */
+
+package io.hops.hopsworks.api.tensorflow;
+
+import javax.xml.bind.annotation.XmlEnum;
+
+@XmlEnum
+public enum TfServingLifecycleView {
```
[the remark this answers is not in the document]
Comment (SirOibaf) [reply hopsworks-api/src/main/java/io/hops/hopsworks/api/tensorflow/TfServingLifecycleView.java]: This class was used in an early implementation. Forgot to get rid of it

File: hopsworks-api/src/main/java/io/hops/hopsworks/api/project/ProjectService.java [hunk 3]
```diff
@@ -923,16 +923,25 @@ public JupyterService jupyter(
...
+      @PathParam("id") Integer id,
+      @Context SecurityContext sc) throws AppException {
     Project project = projectController.findProjectById(id);
     if (project == null) {
       throw new AppException(Response.Status.BAD_REQUEST.getStatusCode(),
           ResponseMessages.PROJECT_NOT_FOUND);
     }
-    this.tfServingService.setProjectId(id);
+
+    Users user = userFacade.findByEmail(sc.getUserPrincipal().getName());
```
[the remark this answers is not in the document]
Comment (SirOibaf) [reply hopsworks-api/src/main/java/io/hops/hopsworks/api/project/ProjectService.java]: TBH I copy pasted from another method. However we are not consistent. Sometimes with fetch the user and check that it's not null, sometimes we don't. I guess one of the 2 is wrong.

[the remark this answers is not in the document]
Comment (tkakantousis) [reply hopsworks-api/src/main/java/io/hops/hopsworks/api/project/ProjectService.java]: Sure, in v2 we will do it consistently

File: hopsworks-api/src/main/java/io/hops/hopsworks/api/tensorflow/ServingConfView.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (tkakantousis) [reply hopsworks-api/src/main/java/io/hops/hopsworks/api/tensorflow/ServingConfView.java]: Ok thanks, we could rename `ServingConf ` to `ServingConfResource` and the `ServingConfView` to `ServingConf`. It dosn't need to be called a view. If some fields are null, they are not returned to the client so we don't need a view for every possible combination of returned objects.

File: hopsworks-api/src/main/java/io/hops/hopsworks/api/project/ProjectService.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (ErmiasG) [reply hopsworks-api/src/main/java/io/hops/hopsworks/api/project/ProjectService.java]: I guess it is redundant.

[the remark this answers is not in the document]
Comment (SirOibaf) [reply hopsworks-api/src/main/java/io/hops/hopsworks/api/project/ProjectService.java]: Fixed

**Comment by SirOibaf** [comment]
Testing https://github.com/logicalclocks/hops-testing/pull/142

**Pull request merged**